### DISCUSSION TITLE
What paths are written to by the installer?

### DISCUSSION THREAD

**Discussion by WilliamDEdwards** [composer/composer#11083] [Q&A / Support] [answered]
What files does the [installer](https://getcomposer.org/installer) write to? I.e. which files should be removed to 'clean' Composer from a system?

I read the installation script. It seems like it writes to:

- The resulting PHAR file inside `--install-dir`
- `.config/composer/keys.tags.pub` and `.config/composer/keys.dev.pub` (or an alternative path returned by the `getHomeDir` function)

Did I miss any paths?

**Comment by Seldaek** [comment, marked as answer]
Nope I think that's about it. Generally speaking `~/.config/composer` and `~/.cache/composer` you might want to delete to remove any machine-wide traces of composer (at least the cache one is for sure harmless).
